### PR TITLE
DM-38738: Merge dev with main for rubin-env 6.0.0.

### PR DESCRIPTION
**Pull request by ktlim**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by ktlim** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -146,18 +143,18 @@ outputs:
         - idds-client >=0.11.0
         - idds-doma >=0.11.0
         - iminuit >=2
-        - lmfit <1.0.3
+        - lmfit
         - lsstdesc.coord
-        - matplotlib-base !=3.6.0,<3.7.0
+        - matplotlib-base !=3.6.0
```
Comment (erykoff) [inline recipe/meta.yaml:148]: Wait, I thought that display_matplotlib was breaking with 3.7.0 and later?  DM-38089

Comment (ktlim) [reply recipe/meta.yaml:148]: Thanks, that's why I wanted a second pair of eyes.  Some of these pins/unpins got messy in the rebase.

**Comment by erykoff** [comment]
@conda-forge-admin, please rerender

**Review by erykoff — APPROVED** [review]
I think this is right. Just making sure we can release the ginga pin on a build?

**Comment by ktlim** [comment]
Yes, since ginga is in rubin-env-rsp, it is always locked to a container, so there are never any compatibility questions.  All we need is a new build to get it to be built into later Science Platform containers.

**Pull request merged**